src/app/static/src/app/components/plots/utils.ts

```diff
@@ -136,8 +135,17 @@ export const iterateDataValues = function(
     func: (areaId: string,
            indicatorMeta: ChoroplethIndicatorMetadata, value: number) => void) {
 
+    const selectedFilterValueIds: Dict<string[]> = {};
+    const validFilters = filters && filters.filter(f => f.options && f.options.length > 0);
+
+    if (validFilters && selectedFilterValues) {
+        for (const f of validFilters) {
+            selectedFilterValueIds[f.id] = selectedFilterValues[f.id].map(n => n.id)
```

Comment:
I get an error on this line: `TypeError: Cannot read property 'map' of undefined` when, having just arrived in the Output step, I set the colour scale for the default indicator to 'Filtered dataset' then switch to the Bubble plot. I guess the selectedFilterValues in the BubblePlot haven't yet been initialised when that  code is first called.